dcos/package.py

```diff
@@ -24,28 +24,29 @@ def uninstall(pkg, package_name, remove_all, app_id, cli, app):
     :rtype: None
     """
 
-    if cli is False and app is False:
-        cli = app = True
-
     uninstalled = False
     installed = installed_packages(
-        pkg, app_id, package_name, cli_only=False)
-    installed_cli = next((True for installed_pkg in installed
-                          if installed_pkg.get("command")), False)
-    installed_app = next((True for installed_pkg in installed
-                          if installed_pkg.get("apps")), False)
+        pkg, None, package_name, cli_only=False)
+    installed_pkg = next(iter(installed), None)
+    installed_cli = installed_pkg.get("command") if installed_pkg else None
+    installed_app = installed_pkg.get("apps") if installed_pkg else []
 
-    if cli and installed_cli:
-        if subcommand.uninstall(package_name):
-            uninstalled = True
+    if cli is False and app is False:
+        app = True
```

Comment:
It's not clear to me what these lines of code are intended to do. Why do we check if `app` is false, and if so, set it immediately to true? I would think that passing in `app` or `cli` as False would mean that we explicitly didn't want to uninstall one or the other of these. We either need a comment or some better naming of things to make what is going on here more apparent.

Reply:
basically those variables are tied to command line options, having both to false means the user didn't pass `--cli` nor `--app`, in that situation we are trying to uninstall both if possible, otherwise fallback to app uninstall only.

Reply:
We should definitely have a comment about this here then, because that is not at all obvious.